src/Core/src/Platform/iOS/ResignFirstResponderTouchGestureRecognizer.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	internal class ResignFirstResponderTouchGestureRecognizer : UITapGestureRecognizer
+	{
+		UIView? _targetView;
+		Token? _token;
+
+		public ResignFirstResponderTouchGestureRecognizer(UIView targetView) :
+			base()
+		{
+			ShouldRecognizeSimultaneously = (recognizer, gestureRecognizer) => true;
+			ShouldReceiveTouch = OnShouldReceiveTouch;
+			CancelsTouchesInView = false;
+			DelaysTouchesEnded = false;
+			DelaysTouchesBegan = false;
+
+			_token = AddTarget((a) =>
+			{
+				if (a is ResignFirstResponderTouchGestureRecognizer gr && gr.State == UIGestureRecognizerState.Ended)
+				{
+					gr.OnTapped();
+				}
+			});
+
+			_targetView = targetView;
+		}
+
+		void OnTapped()
+		{
+			if (_targetView?.IsFirstResponder == true)
+				_targetView?.ResignFirstResponder();
+
+			Disconnect();
+		}
+
+		internal void Disconnect()
+		{
+			if (_token != null)
+				RemoveTarget(_token);
+
+			_token = null;
+			_targetView = null;
+		}
+
+		bool OnShouldReceiveTouch(UIGestureRecognizer recognizer, UITouch touch)
+		{
+			foreach (UIView v in ViewAndSuperviewsOfView(touch.View))
+			{
+				if (v != null && (v is UITableView || v is UITableViewCell || v.CanBecomeFirstResponder))
+					return false;
+			}
+
+			return true;
+		}
+
+		IEnumerable<UIView> ViewAndSuperviewsOfView(UIView view)
+		{
+			while (view != null)
+			{
+				yield return view;
+				view = view.Superview;
+			}
+		}
+
+		internal static void Update(UITextView uIControl, UIWindow? window)
+		{
+			if (window != null)
+			{
+				uIControl.Started += OnEditingDidBegin;
+				uIControl.Ended += OnEditingDidEnd;
+			}
+			else
+			{
+				uIControl.Started -= OnEditingDidBegin;
+				uIControl.Ended -= OnEditingDidEnd;
+			}
+		}
+
+		internal static void Update(UIControl uIControl, UIWindow? window)
```

Comment:
trivial, but I think the `uIControl` casing can be a bit confusing

```suggestion
		internal static void Update(UIControl platformControl, UIWindow? window)
```